jl/utils.jl

```diff
@@ -0,0 +1,189 @@
+type Point
+    line::Int
+    column::Int
+end
+
+type Range
+    start::Point
+    stop::Point
+end
+
+function pos2point(offsets::Vector{Int}, pos::Int)
+    line = searchsortedfirst(offsets, pos)
+    column = line > 1 ? pos - offsets[line-1] : pos
+    return Point(line, column)
+end
+
+function readfirstlines(fpath::AbstractString, n::Int)
+    code = readstring(fpath)
+    if isempty(code)
+       return String[]
+    end
+    count = 1
+    pos = 1
+    while pos < sizeof(code) && count < n
+       if code[pos] == '\n'
+           count += 1
+        end
+        pos += sizeof(string(code[pos]))
+    end
+
+    str = code[1:pos]
+    return split(str, '\n')
+end
+
+function isimportcmd(cmd::AbstractString)
+    return startswith(cmd, "using") || startswith(cmd, "import")
+end
+
+function doimport!(cmdcache::Set{AbstractString}, filename::AbstractString)
+    lines = readfirstlines(filename, 20)
```

Comment:
This is code I didn't write/change. I was going to change it to simply search for all instances of using/import/include to ensure all relevant modules get included into the namespace.